leaderelection/src/main/java/io/scalecube/leaderelection/MessageListener.java

```diff
@@ -0,0 +1,100 @@
+package io.scalecube.leaderelection;
+
+import io.scalecube.cluster.ICluster;
+import io.scalecube.transport.Message;
+
+import rx.functions.Action1;
+import rx.functions.Func1;
+
+/**
+ * Created by ronenn on 9/13/2016.
+ */
+public abstract class MessageListener {
+
+  enum ListenType {
+    GOSSIP, TRANSPORT, GOSSIP_OR_TRANSPORT
+  }
+  
+  public final ICluster cluster;
+
+  public MessageListener(ICluster cluster) {

```

Comment:
the idea is to listen to either gossip or message hence message listener
the idea is to have simple code that does both and nor have the verbose filer function as it will explode the code for each message type

i understand there is code duplication and maybe this can be imoroved